lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -4023,9 +4023,10 @@ def as_cardan_angles(self):
         The inverse of `from_cardan_angles()`.
         Note that the angles returned are in radians, not degrees.
         """
-        qw = self.scalar
-        qx, qy, qz = self.vector[..., :]
+        n = np.sqrt(self.norm)
+        qw = self.scalar / n
+        qx, qy, qz = self.vector[..., :] / n
         azim = np.arctan2(2*(-qw*qz+qx*qy), qw*qw+qx*qx-qy*qy-qz*qz)
-        elev = np.arcsin( 2*( qw*qy+qz*qx)/(qw*qw+qx*qx+qy*qy+qz*qz))  # noqa E201
+        elev = np.arcsin(np.clip(2*(qx*qz + qw*qy), -1, 1))
```

Comment:
But why (the np.clip)? I have a hard time imagining how you'd possibly ever get the arcsin argument out of range...

Reply:
There were floating point round-off errors on the macos github CI runners, which was resulting in values inside the expression of (eg) 1.000000001 and leading to an error being thrown here. I'm not sure of a cleaner robust way to do this than clipping after norming above, but am open to suggestions.

Reply:
All right, touché. I apologize for being so curt yesterday (it was not a good day for me). Let's see if we can make it work as good as we can.
Is there any chance we can extract the qw, qx, qy, and qz from the macos runners where the error shows up? And some of the intermediate results? Hopefully that will give some insight as to why we're confronted with this. (What hardware is used to do the calculations, what processor?)

Reply:
Floating point inaccuracies are something that pop up fairly frequently for matplotlib's tests, and unfortunately we won't ever get perfectly reproducible results across architectures. It's why we use `assert_allclose` a lot of places in testing. That doesn't work in cases like this where you are close to a discrete threshold however, hence the clipping. The good news is that in 99% of cases for matplotlib, the end result is visual so floating point inaccuracies don't matter to the end user.

Reply:
Well I understand the expediency of adding an `np.clip()` and moving on, it gets this job done; but I thought we had the [IEEE 754 Standard for Floating-Point Arithmetic](https://en.wikipedia.org/wiki/IEEE_754) for quite a while now, to help make things like this reliable and portable. And it is not like we're doing something arcane: just addition, multiplication and division, and not even a whole lot of it. So I'm somewhat surprised by the incompatibility (why only on macos?). Seems you're on to something.